ui/de-lib/src/main/java/org/iplantc/de/admin/desktop/client/ontologies/presenter/OntologiesPresenterImpl.java

```diff
@@ -535,4 +568,62 @@ public void onSuccess(Void result) {
     void displayErrorToAdmin() {
         ErrorHandler.post(appearance.ontologyAttrMatchingError());
     }
+
+    @Override
+    public void onRestoreAppButtonClicked(RestoreAppButtonClicked event) {
+        final App app = event.getApp();
+        Preconditions.checkNotNull(app);
+        Preconditions.checkArgument(app.isDeleted());
+
+        view.maskGrids(appearance.loadingMask());
+        adminAppService.restoreApp(app, new AsyncCallback<App>() {
+            @Override
+            public void onFailure(Throwable caught) {
+                view.unmaskGrids();
+                JSONObject obj = JSONParser.parseStrict(caught.getMessage()).isObject();
+                String reason = jsonUtil.trim(obj.get("reason").toString());
```

Comment:
In fact, if we're just going to be parsing a JSONObject here, then we should use `JsonUtil.getObject` and `JsonUtil.getString`, which will do the null-checking for us.